src/pages/register/index.js

```diff
@@ -6,49 +6,86 @@ import CredentialsCard from "../../components/credentials";
 import "./register.css";
 
 const Register = () => {
-	const { onRegister } = useAuth();
-	const [formData, setFormData] = useState({ email: "", password: "" });
+  const { onRegister } = useAuth();
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [errorMsg, setErrorMsg] = useState("");
+  const specialChars = /^(?=.*[!@#$%^&*])/;
+  const numbers = /^(?=.*[0-9])/;
+  // This doesn't work and could be implemented later
+  // const forbiddenChars = /^(?=.*[(){}[]|`¬¦!"£$%^&*"<>:;#~_-+=,@])/;
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
 
-	const onChange = (e) => {
-		const { name, value } = e.target;
-		setFormData({ ...formData, [name]: value });
-	};
+  function checkPasswordAndSubmit() {
+    let enteredPass = formData.password;
+    setErrorMsg("");
+    if (enteredPass.length < 8) {
+      setErrorMsg("Password must be at least 8 characters long!");
+      return false;
+    } else if (
+      enteredPass.toLowerCase() === enteredPass ||
+      enteredPass.toUpperCase() === enteredPass
+    ) {
+      setErrorMsg(
+        "Password must contain at least 1 uppercase and 1 lowercase letter"
+      );
+      return false;
+    } else if (!numbers.test(enteredPass)) {
+      setErrorMsg("Password must contain at least 1 number");
+      return false;
+    } else if (!specialChars.test(enteredPass)) {
+      setErrorMsg("Password must contain at least 1 special character");
+      return false;
+    }
+    // This does not work and should be fixed later
+    // if (forbiddenChars.test(enteredPass)) {
+    //   setErrorMsg("Password contains forbidden characters!");
+    // } else
+    else {
+      onRegister(formData.email, formData.password);
```

Comment:
i'd add a comment here above onRegister giving a short explanation that onRegister takes care of submitting the registration for the user... however, this function should return something so that this component can display an error in case that onRegister fails? what do you think?